### PR TITLE
WIP: Lab_08_01 - Resource management solve lab

### PR DESCRIPTION
**Pull request by gmontalvoy**
Added files:

- lab_08_01 (d)
- solve_lab.yml
- Initial version to resource management lab

### CODE REVIEW

**Comment by stencell** [comment]
Overall, it is okay. I would like to see some checks rather than best effort guesses at how long to pause. Those just cause problems. See something like this for an example that is pretty easy to implement:

https://github.com/redhat-cop/agnosticd/blob/development/ansible/roles_ocp_workloads/ocp4_workload_rhacm/tasks/workload.yml#L82-L95

**Comment by gmontalvoy** [comment]
- Changed best-effort to k8s_facts tasks.

File: courses/ocp4_advanced_deployment/lab_08_1/solve_lab.yml [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+
+    - name: Wait until custom adapter is available
+      k8s_facts:
+        api_version: apps/v1
+        kind: Deployment
+        name: custom-metrics-apiserver
+        namespace: my-prometheus
+      register: r_custom_adapter
+      until:
+      - r_custom_adapter | length > 0
```
Comment (stencell) [inline courses/ocp4_advanced_deployment/lab_08_1/solve_lab.yml:208]: I think this is a weak check. You are looking to see if a Deployment exists, but not the resulting objects (i.e. Pod(s)) that the Deployment is creating or the functionality they provide. As a solver, each of these tasks should be as close as possible to the task that a user performs. In the current version of this lab, they would be running this command:
```
oc get --raw /apis/custom.metrics.k8s.io/v1beta1/ | jq -r '.resources[] | select(.name | contains("pods/http"))'
```
This should be what you are checking to make sure the result is as expected.

Comment (gmontalvoy) [reply courses/ocp4_advanced_deployment/lab_08_1/solve_lab.yml:208]: Sounds reasonable let me figure out how to implement this, thanks for pointing

**Comment by stencell** [comment]
I made another comment on the check to address.

In addition, as a solver, you need to clean up as a user would be expected to in the lab. In the current version, they delete the resource & custom HPAs when done. The solver would need to do this as well.

This is the kind of lab that we would never actually have a user execute the solver for. There is nothing that happens in this lab that is required to to be completed for the rest of the labs to function. It is, however, a great thing to have in order to automate QA.

File: courses/ocp4_advanced_deployment/lab_08_1/solve_lab.yml (hunk 1 above)
Comment (gmontalvoy) [reply courses/ocp4_advanced_deployment/lab_08_1/solve_lab.yml:208]: Resolved on next commit.

**Comment by gmontalvoy** [comment]
Regarding service check, please take note I am not looking for a deployment, I am looking for an existing service to expose it creating a route as lab instruction said.

- Changed {{ r_custom_adapter }} to wait until k8s adapter is available before moving on.

A reset playbook will follow.

IMHO, I don't see the point to clear everything at the last steps of this playbook, it will cause nobody will see anything done in lab, students forced to use this playbook should see the result by themselves before clear everything, it is not a dependency for other labs but keep resources existing is not harmful for them.

**Review by stencell — APPROVED** [review]
Looks fine now. Please clean up the commented out code, though.

**Comment by varodrig** [comment]
@gmontalvoy  if this includes version OCP 4.6 it shouldn't being merged until all the modules are migrated and tested to 4.6.

**Comment by gmontalvoy** [comment]
Ok, im done.

Thanks

**Comment by wkulhanek** [comment]
Absolutely. I'm not sure we need to close it. Just rename it to WIP to say it's work in progress. And when we merge the lab instructions we can merge the FTL. @gmontalvoy @varodrig

**Comment by wkulhanek** [comment]
Is this dependent on 4.6? If so we can't merge. We need the whole class on 4.6. If it works on 4.5 we could merge - but would need the module updated, too.

**Pull request closed without merging**